src/analysis/inner_product_space/finite_dimensional.lean

```diff
@@ -0,0 +1,351 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import analysis.inner_product_space.adjoint
+import analysis.inner_product_space.spectrum
+
+/-!
+# Finite-dimensional inner product spaces
+
+In this file, we prove some results in finite-dimensional inner product spaces.
+
+## Notation
+
+This file uses the local notation `P _` for `orthogonal_projection _`
+and `↥P _` for the extended orthogonal projection `V →L[ℂ] V`.
+
+We let `V` be an inner product space over `ℂ`.
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+local notation `P` := orthogonal_projection
+
+/-- `U` is `T` invariant: `∀ u : V`, if `u ∈ U` then `T u ∈ U`-/
+def submodule.invariant (U : submodule ℂ V) (T : V →ₗ[ℂ] V) : Prop := U ≤ U.comap T
+lemma submodule.invariant_iff (U : submodule ℂ V) (T : V →ₗ[ℂ] V) :
+  submodule.invariant U T ↔ T '' U ⊆ U :=
+by simp only [set.image_subset_iff]; refl
+
+/-- `↥P _` is the extended version of `P _` -/
+noncomputable def orthogonal_projection.extend [finite_dimensional ℂ V]
+  (U : submodule ℂ V) : V →L[ℂ] V :=
+linear_map.mk_continuous
+  { to_fun := by intro v; exact ↑(P U v),
+    map_add' := by intros; rw [map_add, submodule.coe_add],
+    map_smul' := by intros; rw [map_smul, submodule.coe_smul_of_tower, ring_hom.id_apply] }
+-- the following is copied from `def orthogonal_projection` in
+-- `analysis/inner_product_spaces/projection`
+  1 (λ x, begin simp only [one_mul, linear_map.coe_mk],
+     refine le_of_pow_le_pow 2 (norm_nonneg _) (by norm_num) _,
+     change ‖orthogonal_projection_fn U x‖ ^ 2 ≤ ‖x‖ ^ 2,
+     nlinarith [orthogonal_projection_fn_norm_sq U x] end)
```

Comment:
This definition is just `(orthogonal_projection U).comp subtypeL`; so I don't think it it is needed.